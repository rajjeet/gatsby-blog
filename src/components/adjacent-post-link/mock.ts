export const previous = {
  fields: { slug: '/previous/' },
  frontmatter: { title: 'Introducing the New React Concurrent Mode' },
};
export const next = {
  fields: { slug: '/next/' },
  frontmatter: { title: 'The Implications of React Concurrent Mode' },
};
