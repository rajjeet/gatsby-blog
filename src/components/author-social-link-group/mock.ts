import { TProps } from './types';

export const makeProps = (): TProps => ({
  className: '',
});

