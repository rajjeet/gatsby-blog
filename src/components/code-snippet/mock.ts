import { TProps } from './types';

export const makeProps = (): TProps => ({
  children: `
const mainReducer = (state, action) => {
  console.log('Before', state);
  console.log('Action', action);
  let result = {
    count: countReducer(state.count, action)      
  };
  console.log('After', result);
  return result
};      
`,
  fileName: 'index.js',
});
