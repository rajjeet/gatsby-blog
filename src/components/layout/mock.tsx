import React from 'react';
import { TProps } from './types';
import { createMockGatsbyImageSharpFluid } from '../../utils/testing';
import { siteMetadata } from '../../../gatsby-config';

export const makeProps = (): TProps => ({
  children: <div>Sample child</div>,
  data: {
    file: createMockGatsbyImageSharpFluid.file,
    site: siteMetadata,
  },
});

