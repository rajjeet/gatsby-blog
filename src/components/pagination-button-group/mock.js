export const makeProps = () => ({
  className: {}, currentPage: 1, numOfPages: 5, paginationSlug: '',
});
