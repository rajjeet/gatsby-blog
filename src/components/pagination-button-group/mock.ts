export const makeProps = ({ currentPage = 1, numOfPages = 5 } = {}): TProps => ({
  className: '', currentPage, numOfPages, paginationSlug: '',
});

export type TProps = {
  className: string;
  currentPage: number;
  numOfPages: number;
  paginationSlug: string;
};
