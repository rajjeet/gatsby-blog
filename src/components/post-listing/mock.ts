import { TProps } from './types';
import { mockPosts } from '../../fixtures/posts';

export const makeProps = ({
  posts = mockPosts,
  heading = 'Post Listing',
  numOfPages = 5,
  currentPage = 1,
  paginationSlug = '/sample_slug/',
} = {}): TProps => ({
  posts,
  heading,
  numOfPages,
  currentPage,
  paginationSlug,
});

