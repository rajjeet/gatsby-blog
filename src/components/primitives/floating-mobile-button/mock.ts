import { faTimes } from '@fortawesome/free-solid-svg-icons';

export const makeProps = () => ({
  icon: faTimes,
});
