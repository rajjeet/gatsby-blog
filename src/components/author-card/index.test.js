import React from 'react';
import { StaticQuery } from 'gatsby';
import { render, cleanup } from '@testing-library/react';
import AuthorCard from './index';
import { createMockGatsbyImageSharpFluid } from '../../utils/testing';
import { makeProps } from './mock';

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render: renderQuery }) => renderQuery({
    file: createMockGatsbyImageSharpFluid.file,
  }));
});

afterEach(cleanup);

describe('<AuthorCard />', () => {
  it('should render', () => {
    const { asFragment } = render(<AuthorCard {...makeProps()} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should have an image of author headshot', () => {
    const { getByAltText } = render(<AuthorCard {...makeProps()} />);
    expect(getByAltText(/author.*headshot/i)).toBeDefined();
  });

  it('should have a heading for author name', () => {
    const { getByText } = render(<AuthorCard {...makeProps()} />);
    expect(getByText('Rajjeet Phull')).toBeDefined();
  });

  it('should have a description specifying occupation of author as software developer', () => {
    const { getByText } = render(<AuthorCard {...makeProps()} />);
    expect(getByText(/Software Developer/)).toBeDefined();
  });
});

