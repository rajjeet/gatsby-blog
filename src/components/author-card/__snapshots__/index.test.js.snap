// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthorCard /> should render 1`] = `
<DocumentFragment>
  <div
    class="[object Object] sc-bwzfXH bBObBL"
  >
    <div
      class="sc-bdVaJa erGNdq gatsby-image-wrapper"
      style="position: relative; overflow: hidden;"
    >
      <div
        style="width: 100%; padding-bottom: 100%;"
      />
      <picture>
        <source
          sizes=""
          srcset=""
        />
        <img
          alt="Author headshot"
          src=""
          style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 0.5s;"
        />
      </picture>
      <noscript>
        &lt;picture&gt;&lt;img src="" alt="Author headshot" style="position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
      </noscript>
    </div>
    <h3>
      Rajjeet Phull
    </h3>
    <p>
      Software Developer. Specializing in .NET(C#), React, SQL Server, and AWS.
    </p>
  </div>
</DocumentFragment>
`;
